function/run.go

```diff
@@ -19,9 +17,9 @@ type RunRequest struct {
 // RunResponse represents a response to a RunRequest. An instance of this
 // response struct is supplied as an argument to the Function type Run method.
 type RunResponse struct {
-	// Diagnostics report errors or warnings related to defining the function.
-	// An empty slice indicates success, with no warnings or errors generated.
-	Diagnostics diag.Diagnostics
+	// Errors contains errors related to running the function.
+	// A nil error indicates success, with no errors generated.
+	Errors fwerror.FunctionErrors
```

Comment:
Would it be safer for the framework to expose this abstraction more inline with the protocol (e.g. a single function error) and then potentially offer opinionated helper(s) that may change over time for joining multiple errors into one? Is `errors.Join()` or our own version of that acceptable enough? We have tried to avoid the framework from forcing opinionated behaviors on provider developer responses where possible in the past.

Reply:
As mentioned above, we now have a single `function.FuncError` struct type, with `function.ConcatFuncErrors` and `function.FuncErrorFromDiags` helper functions. The `function.ConcatFuncErrors` is our current version of `errors.Join()`.